### PR TITLE
Add search for account by alias or EVM address to GraphQL API

### LINKED ISSUE #5318: GraphQL Search for account by alias or address
### Problem

As a consumer of the mirror node [GraphQL API](https://github.com/hashgraph/hedera-improvement-proposal/pull/668), I'd like to be able to lookup accounts by either its unique alias or EVM address.

### Solution

* Add support for `account(id: {alias})` and `account(id: {evmAddress})` lookup. Execute the same entity lookup as `/api/v1/accounts/{idOrAliasOrEvmAddress}` does.
* Add unit/integration tests for the above
* Add a query to postman

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by syed-bcw**
**Description**:

- Added support for `account(id: {alias})` and `account(id: {evmAddress})` lookup. Executes the same entity lookup as `/api/v1/accounts/{idOrAliasOrEvmAddress}` does.
- Added unit/integration tests for the above
- Added a query to postman

**Related issue(s)**:

Fixes #5318 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/controller/AccountControllerTest.java [hunk 1]
```diff
@@ -34,6 +35,10 @@
 import com.hedera.mirror.graphql.GraphqlIntegrationTest;
 import com.hedera.mirror.graphql.mapper.AccountMapper;
 import com.hedera.mirror.graphql.viewmodel.Account;
+import org.apache.commons.codec.binary.Base32;
+
+import java.math.BigInteger;
+import java.util.concurrent.TimeUnit;
```
Comment (edwin-greene) [inline hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/controller/AccountControllerTest.java:41]: Looks like TimeUnit isn't used. Can you optimize imports? It should place the Hex import below the license as well.

File: hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/controller/AccountControllerTest.java [hunk 2]
```diff
@@ -74,6 +77,11 @@ void invalidInput(String query, String error) {
                 );
     }
 
+    @Test
+    void invalidEvmAddress(){
+
+    }
+
```
Comment (edwin-greene) [inline hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/controller/AccountControllerTest.java:84]: Can remove or add an invalid evm address test.

File: hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/util/GraphQlUtilsTest.java
Comment (edwin-greene) [inline hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/util/GraphQlUtilsTest.java:1]: Can you add a test that documents what happens when `decodeEvmAddress` is called with an empty string and null?

File: hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/util/GraphQlUtilsTest.java [hunk 3]
```diff
@@ -85,4 +88,40 @@ void validateOneOf() {
...
+        Base32 base32 = new Base32();
+        String alias = "Account1";
+        var node = Account.builder().withAlias(base32.encodeAsString(alias.getBytes())).build();
+        byte[] decodedAlias = GraphQlUtils.decodeBase32(node.getAlias());
+        assertThat(decodedAlias).isEqualTo(alias.getBytes());
+    }
+
+    @Test
+    void decodeEvmAddress() {
+        String evmAddress = "0000000000000000000000000000000000000001";
```
Comment (edwin-greene) [inline hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/util/GraphQlUtilsTest.java:103]: Should this also work with `0x` prefixed addresses? If so, we should have a test with an input like  `0x00000000000000000000000000000000000000001`.

File: hedera-mirror-graphql/src/main/java/com/hedera/mirror/graphql/controller/AccountController.java [hunk 4]
```diff
@@ -62,6 +66,41 @@ Mono<Account> account(@Argument @Valid AccountInput input) {
...
+                    account = entityService.getByIdAndType(entity, EntityType.ACCOUNT)
+                            .map(accountMapper::map);
+                }
+            }
+
+            return Mono.justOrEmpty(account);
+        }
+
+        if (evmAddress != null) {
+            byte[] evmBytes = decodeEvmAddress(evmAddress);
```
Comment (steven-sheehy) [inline hedera-mirror-graphql/src/main/java/com/hedera/mirror/graphql/controller/AccountController.java:88]: One new special case for [HIP-583](https://github.com/hashgraph/hedera-improvement-proposal/blob/1cb3c7ce6c19874caa80c667ad33fd9c8129dd38/HIP/hip-583.md) that we're tackling concurrently in #5507 is that we should check if the EVM address is an entity ID encoded into an EVM address. This occurs when the leftmost 12 bytes are zero. If so, we can directly convert to EntityID and skip the EVM address lookup.

File: hedera-mirror-graphql/src/main/java/com/hedera/mirror/graphql/controller/AccountController.java [hunk 5]
```diff
@@ -62,6 +66,41 @@ Mono<Account> account(@Argument @Valid AccountInput input) {
...
+                    .map(accountMapper::map);
+
+            if (account.isEmpty()) {
+                var entity = fromEvmAddress(evmBytes);
+                if (entity != null) {
+                    account = entityService.getByIdAndType(entity, EntityType.ACCOUNT)
+                            .map(accountMapper::map);
+                }
+            }
+            return Mono.justOrEmpty(account);
```
Comment (steven-sheehy) [inline hedera-mirror-graphql/src/main/java/com/hedera/mirror/graphql/controller/AccountController.java:101]: Remove the fallback behavior.
```suggestion
            return Mono.justOrEmpty(entityService.getByEvmAddress(evmBytes, EntityType.ACCOUNT)
                            .map(accountMapper::map));
```

File: hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/controller/AccountControllerTest.java [hunk 6]
```diff
@@ -84,6 +92,26 @@ void missing() {
                 .valueIsNull();
     }
 
+    @Test
+    void missingEvmAddress() {
```
Comment (steven-sheehy) [inline hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/controller/AccountControllerTest.java:96]: Recommend making `missing()` a parameterized test and test each input format.

File: hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/controller/AccountControllerTest.java [hunk 7]
```diff
@@ -121,6 +149,158 @@ query Account($id: Long!) {
                 .satisfies(a -> assertThat(a).usingRecursiveComparison().isEqualTo(accountMapper.map(entity)));
     }
 
+    @Test
+    void successByEvmAddress() {
```
Comment (steven-sheehy) [inline hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/controller/AccountControllerTest.java:153]: Should make parameterized with two booleans: one to control 0x prefix and another to control lowercase vs uppercase.

File: charts/hedera-mirror-graphql/postman.json [hunk 8]
```diff
@@ -1,165 +1,354 @@
...
+					"setVariableEverywhere(\"default_evm_address\", default_evm_address)",
+					"setVariableEverywhere(\"default_alias\", default_alias)",
+					"",
+					"",
+					"function setVariableEverywhere(key, value) {",
+					"    pm.variables.set(key, value);",
+					"    pm.globals.set(key, value);",
+					"    pm.collectionVariables.set(key, value)",
+					"}",
+					"",
```
Comment (steven-sheehy) [inline charts/hedera-mirror-graphql/postman.json:273]: Think all this logic is removed if fallback alias to entity ID goes away.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: charts/hedera-mirror-graphql/postman.json [hunk 9]
```diff
@@ -1,165 +1,354 @@
...
+			"key": "default_account",
+			"value": "98"
+		},
+		{
+			"key": "baseUrl",
+			"value": "http://localhost:8083"
+		},
+		{
+			"key": "default_evm_address",
+			"value": "0000000000000000000000000000000000000062"
```
Comment (steven-sheehy) [inline charts/hedera-mirror-graphql/postman.json:347]: These new EVM address and alias tests will fail in CI since they won't exist in the demo bucket. We should default these variables to empty and only verify the results if they are non-empty. See https://medium.com/apis-with-valentine/how-to-reuse-tests-and-skip-collection-tests-in-postman-da47c50444c8

**Comment by syed-bcw** [comment]
@edwin-greene , @steven-sheehy - Addressed all changes, need rereview 😊

File: hedera-mirror-graphql/src/main/java/com/hedera/mirror/graphql/util/GraphQlUtils.java [hunk 10]
```diff
@@ -94,4 +102,26 @@ public static void validateOneOf(Object... values) {
...
+            return null;
+        }
+        if (evmAddress.startsWith(HEX_PREFIX)) {
+            evmAddress = StringUtils.removeStart(evmAddress, HEX_PREFIX);
+        }
+        try {
+            return Hex.decodeHex(evmAddress);
+        } catch (DecoderException e) {
+            throw new IllegalArgumentException("Must provide proper evmAddress but (" + evmAddress + ") have been " +
+                    "provided");
```
Comment (steven-sheehy) [inline hedera-mirror-graphql/src/main/java/com/hedera/mirror/graphql/util/GraphQlUtils.java:124]: nit: could simplify
```suggestion
            throw new IllegalArgumentException("Unable to decode evmAddress: " + evmAddress);
```

File: hedera-mirror-graphql/src/main/java/com/hedera/mirror/graphql/controller/AccountController.java (hunk 4 above)
Comment (steven-sheehy) [reply hedera-mirror-graphql/src/main/java/com/hedera/mirror/graphql/controller/AccountController.java]: This logic was missed. Should be done in service layer.

Comment (syed-bcw) [reply hedera-mirror-graphql/src/main/java/com/hedera/mirror/graphql/controller/AccountController.java]: I thought that I can skip this logic check since this will be handled by HIP-583, let me add that check back in service layer

Comment (syed-bcw) [reply hedera-mirror-graphql/src/main/java/com/hedera/mirror/graphql/controller/AccountController.java]: I thought that I can skip this logic check since this will be handled by HIP-583, let me add that check back in service layer

Comment (syed-bcw) [reply hedera-mirror-graphql/src/main/java/com/hedera/mirror/graphql/controller/AccountController.java]: @steven-sheehy - made recommended changes 😊

File: hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/util/GraphQlUtilsTest.java
Comment (syed-bcw) [reply hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/util/GraphQlUtilsTest.java:1]: @edwin-greene - Done 😊

**Review by steven-sheehy — DISMISSED** [review]
LGTM! Thanks for your contribution 🙏🏻

**Review by edwin-greene — DISMISSED** [review]
Looks good!

**Comment by steven-sheehy** [comment]
@syed-bcw Whoops, just noticed that all of the commits aren't GPG [signed](https://docs.github.com/articles/about-gpg/). Do you mind fixing that? You're welcome to squash commits beforehand if it makes it easier.

**Comment by syed-bcw** [comment]
> @syed-bcw Whoops, just noticed that all of the commits aren't GPG [signed](https://docs.github.com/articles/about-gpg/). Do you mind fixing that? You're welcome to squash commits beforehand if it makes it easier.

@steven-sheehy - verified all commits, see if in good state 😊

**Review by steven-sheehy — APPROVED** [review]
